packages/cli/src/services/cache.service.ts

```diff
@@ -0,0 +1,167 @@
+			throw new Error('Cache is disabled');
+		}
+
+		const backend = config.getEnv('cache.backend');
+
+		if (
+			backend === 'redis' ||
+			(backend === 'auto' && config.getEnv('executions.mode') === 'queue')
+		) {
+			// eslint-disable-next-line @typescript-eslint/naming-convention
+			const { redisInsStore } = await import('cache-manager-ioredis-yet');
+
+			// #region TEMPORARY Redis Client Code
+			/*
+			 * TODO: remove once redis service is ready
+			 * this code is just temporary
+			 */
+			// eslint-disable-next-line @typescript-eslint/naming-convention
+			const { default: Redis } = await import('ioredis');
+			let lastTimer = 0;
+			let cumulativeTimeout = 0;
+			const { host, port, username, password, db }: RedisOptions =
+				config.getEnv('queue.bull.redis');
+			const clusterNodes = getRedisClusterNodes();
+			const redisConnectionTimeoutLimit = config.getEnv('queue.bull.redis.timeoutThreshold');
+			const usesRedisCluster = clusterNodes.length > 0;
+			LoggerProxy.debug(
+				usesRedisCluster
+					? `(Cache Service) Initialising Redis cluster connection with nodes: ${clusterNodes
+							.map((e) => `${e.host}:${e.port}`)
+							.join(',')}`
+					: `(Cache Service) Initialising Redis client connection with host: ${
+							host ?? 'localhost'
+					  } and port: ${port ?? '6379'}`,
+			);
+			const sharedRedisOptions: RedisOptions = {
+				username,
+				password,
+				db,
+				enableReadyCheck: false,
+				maxRetriesPerRequest: null,
+			};
+			const redisClient = usesRedisCluster
+				? new Redis.Cluster(
+						clusterNodes.map((node) => ({ host: node.host, port: node.port })),
+						{
+							redisOptions: sharedRedisOptions,
+						},
+				  )
+				: new Redis({
+						host,
+						port,
+						...sharedRedisOptions,
+						retryStrategy: (): number | null => {
+							const now = Date.now();
+							if (now - lastTimer > 30000) {
+								// Means we had no timeout at all or last timeout was temporary and we recovered
+								lastTimer = now;
+								cumulativeTimeout = 0;
+							} else {
+								cumulativeTimeout += now - lastTimer;
+								lastTimer = now;
+								if (cumulativeTimeout > redisConnectionTimeoutLimit) {
+									LoggerProxy.error(
+										`Unable to connect to Redis after ${redisConnectionTimeoutLimit}. Exiting process.`,
+									);
+									process.exit(1);
+								}
+							}
+							return 500;
+						},
+				  });
+			// #endregion TEMPORARY Redis Client Code
+			const redisStore = redisInsStore(redisClient, {
+				ttl: config.getEnv('cache.redis.ttl'),
+			});
+			this.cache = await caching(redisStore);
+		} else {
+			this.cache = await caching('memory', {
+				ttl: config.getEnv('cache.memory.ttl') ?? 10 * 1000,
+				max: config.getEnv('cache.memory.max') ?? 100 * 1024 * 1024,
+			});
+		}
+	}
+
+	async uninit() {
+		if (this.cache) {
+			await this.reset();
+			this.cache = undefined;
+		}
+	}
+
+	getCacheInstance(): RedisCache | MemoryCache | undefined {
+		return this.cache;
+	}
```

Comment:
Maybe we could initialize it if `undefined` in `getCacheInstance` so that we don't have to check for `undefined` at call site.